src/Macros/WeightedRandom.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Spatie\CollectionMacros\Macros;
+
+class WeightedRandom
+{
+    public function __invoke()
+    {
+        return function (callable|string $weightAttribute, $default = null) {
+            if (is_string($weightAttribute)) {
+                $attributeName = $weightAttribute;
+
+                $weightAttribute = function ($item) use ($attributeName) {
+                    return $item[$attributeName];
```

Comment:
I think using `data_get` might be more appropriate here in case the collection consists of objects.

Reply:
Good suggestion, thanks!